osu.Game.Rulesets.Catch/Skinning/Legacy/CatchLegacySkinTransformer.cs

```diff
@@ -65,17 +65,17 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
 
                         return null;
 
-                    case CatchSkinComponents.CatcherIdle:
-                        return this.GetAnimation("fruit-catcher-idle", true, true, true) ??
-                               this.GetAnimation("fruit-ryuuta", true, true, true);
+                    case CatchSkinComponents.Catcher:
+                        // New elements will be ignored when the old element exists.
+                        if (GetTexture(@"fruit-ryuuta") != null ||
+                            GetTexture(@"fruit-ryuuta-0") != null)
+                            return new LegacyCatcherOld();
```

Comment:
In stable, this is actually based on a version check.

```csharp
if (SkinManager.Current.Version < 2.3)
    catcherIdle = catcherFail = catcherKiai = TextureManager.LoadAll(@"fruit-ryuuta");
```

As to whether `FindProvider()` should be used, it depends on what should happen if a skin only provides `fruit-catcher-idle` but not the others. If the default skin should be used as a fallback then this is correct, and if `idle` should be used for the other states then this does need to use `FindProvider()` and the returned skin should be passed into the component and used for lookups. I don't believe there should be any other edge cases.

Reply:
Should version checking logic included here? There is no version checking logic for other rulesets so I'm not sure.
If so, which version should be used, `latest` or default (1.0)?

Indeed, not using `FindProvider` is correct here because stable is falling back to the default skin when only `fruit-catcher-idle` is provided.

Reply:
There are version checks in other rulesets ([example 1](https://github.com/ppy/osu/blob/8a221f501d6ea4411158ca6acec0155eb77e18f2/osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyMainCirclePiece.cs#L164-L168), [example 2](https://github.com/ppy/osu/blob/8a221f501d6ea4411158ca6acec0155eb77e18f2/osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacyInputDrum.cs#L70)). Not super concerned whether one should be used here, just figured I'd bring that info up as it's the ground truth so to speak.

Reply:
Hmm I somehow missed by search.